include/swift/SIL/ApplySite.h

```diff
@@ -729,4 +740,23 @@ template <> struct DenseMapInfo<::swift::FullApplySite> {
 
 } // namespace llvm
 
+//===----------------------------------------------------------------------===//
+//           Inline Definitions to work around Forward Declaration
+//===----------------------------------------------------------------------===//
+
+namespace swift {
+
+inline Optional<FullApplySite> ApplySite::asFullApplySite() const {
```

Comment:
@gottesmm this new API is inconsistent with existing apply site type coercion which always returns an invalid apply site on failure. Adding another layer of Optional is extremely confusing. I don't think you need this API anyway, but it definitely should not change the return type.